Vercel backend dependency erorrs

```
2025-10-04T15:48:06.747Z [error] Cannot find module '@shared/schemas/common'
Require stack:
- /var/task/backend/src/routes/instructorsRouter.js
- /var/task/backend/src/server.js
- /var/task/backend/api/app.js
Did you forget to add it to "dependencies" in `package.json`?
Node.js process exited with exit status: 1. The logs above can help with debugging the issue.
Cannot find module '@shared/schemas/common'
Require stack:
- /var/task/backend/src/routes/instructorsRouter.js
- /var/task/backend/src/server.js
- /var/task/backend/api/app.js
Did you forget to add it to "dependencies" in `package.json`?
Node.js process exited with exit status: 1. The logs above can help with debugging the issue.
```

--- a/backend/.gitignore
+++ b/backend/.gitignore
@@ -1,4 +1,5 @@
 node_modules
+dist
 # Keep environment variables out of version control
 .env
 .vercel

--- a/backend/package.json
+++ b/backend/package.json
@@ -10,6 +10,7 @@
     "prisma:init": "npx prisma migrate reset && npx ts-node ./src/seed/dummy.ts",
     "test": "vitest run",
     "build": "prisma generate && prisma migrate reset --force && echo 'Start executing dummy.ts' && ts-node ./src/seed/dummy.ts && echo 'Finish executing dummy.ts'",
+    "vercel:build": "prisma generate && tsc && tsc-alias && prisma migrate reset --force && echo 'Start executing dummy.ts' && ts-node ./src/seed/dummy.ts && echo 'Finish executing dummy.ts'",
     "lint:unused": "knip"
   },
   "engines": {
@@ -36,6 +37,7 @@
     "prisma": "^5.19.1",
     "supertest": "^7.1.3",
     "ts-node": "^10.9.2",
+    "tsc-alias": "^1.8.16",
     "tsconfig-paths": "^4.2.0",
     "typescript": "^5.4.5",
     "vite": "^7.1.5",

--- a/backend/tsconfig.json
+++ b/backend/tsconfig.json
@@ -2,6 +2,7 @@
   "extends": "@tsconfig/node20/tsconfig.json",
   "compilerOptions": {
     "baseUrl": ".",
+    "outDir": "./dist",
     "paths": {
       "@shared/*": ["../shared/*"]
     }

--- a/backend/vercel.json
+++ b/backend/vercel.json
@@ -1,4 +1,6 @@
 {
   "version": 2,
+  "buildCommand": "npm run vercel:build",
+  "outputDirectory": "dist/backend",
   "rewrites": [{ "source": "/(.*)", "destination": "/api/app" }]
 }

